### PR TITLE
Fixed memory leaks found by 'valgrind --tool=memcheck ntlmssptest'

### PR DESCRIPTION
**Pull request by kvv81**
Signed-off-by: Volodymyr Khomenko <volodymyr@vastdata.com>

### CODE REVIEW

File: src/ntlm_crypto.c [hunk 1]
```diff
@@ -577,6 +577,12 @@ int ntlm_reset_rc4_state(uint32_t flags, bool recv,
     return ret;
 }
 
+void ntlm_release_rc4_state(struct ntlm_signseal_state *state)
```
Comment (simo5) [inline src/ntlm_crypto.c:580]: Can you use this function in gssntlm_delete_sec_context() given you created it ?

**Review by simo5 — CHANGES_REQUESTED** [review]
One nit for consistency, I want to avoid having accessory functions as well as directly freeing state from outside of ntlm_crypto.

Additionally can you split the commit title to just be 'fix memory leaks found by valgrind' and if necessary add more details in the body of the comment?
The current commit title is too long.

File: src/ntlm_crypto.c (hunk 1 above)
Comment (kvv81) [reply src/ntlm_crypto.c:580]: yep, good idea...

**Comment by kvv81** [comment]
Done, please review again :-)

**Review by simo5 — APPROVED** [review]
Thanks, this all looks good!

**Pull request merged**